src/apps/frontend/pages/tasks/task-header.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import toast from 'react-hot-toast';
+
+import { Button, Flex, HeadingLarge } from '../../components';
+import { AsyncError } from '../../types';
+import { ButtonSize } from '../../types/button';
+import { TaskOperationType } from '../../types/task';
+
+import TaskModal from './task-modal';
+import useTaskForm from './tasks-form.hook';
+
+interface TaskHeaderProps {
+  onError?: (error: AsyncError) => void;
+}
+
+const TaskHeader: React.FC<TaskHeaderProps> = ({ onError }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const onSuccess = () => {
+    toast.success('Task has been added successfully');
+    setIsModalOpen(false);
+  };
+
+  const { addTaskFormik, setFormikFieldValue } = useTaskForm({
+    onError,
+    onSuccess,
+  });
+
+  return (
+    <div className="rounded-sm  border border-stroke bg-white p-3 shadow-default">
+      <div className="flex flex-col items-start justify-between gap-5 sm:flex-row sm:items-center ">
+        <Flex>
+          <div className="pl-2">
+            <HeadingLarge>Tasks</HeadingLarge>
+          </div>
+        </Flex>
+        <Flex>
+          <Button
+            onClick={() => setIsModalOpen(!isModalOpen)}
+            size={ButtonSize.COMPACT}
+            startEnhancer={
+              <img src="assets/svg/plus-icon.svg" alt="Plus Icon" />
+            }
+          >
+            Add task
+          </Button>
+          <TaskModal
+            formik={addTaskFormik}
+            setFieldValue={setFormikFieldValue}
+            isModalOpen={isModalOpen}
+            setIsModalOpen={setIsModalOpen}
+            btnText={'Add Task'}
+            taskOperationType={TaskOperationType.ADD}
+          />
+        </Flex>
```

Comment:
<img width="709" alt="Screenshot 2024-03-28 at 11 46 52 AM" src="https://github.com/jalantechnologies/boilerplate-mern/assets/74129325/108e83de-7ffe-4411-82b6-85a0f0044d31">

I think we can refract and remove the Flex here like this, there is no need to wrap the button and modal with flex too, we can simply place the modal outside and wrap the button with div. Please let me know if this is breaking anything